### PR TITLE
One less pickled object in the .hdf5 file

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1474443. The tests are green: https://ci.openquake.org/job/zdevel_oq-risklib/963. This PR requires the companion https://github.com/gem/oq-hazardlib/pull/370

### CODE REVIEW

File: openquake/commonlib/valid.py [hunk 1]
```diff
@@ -857,18 +869,39 @@ class ParamSet(object):
...
+    >>> MyParams.from_(dict(a="'2'", b='7.2'))
+    <MyParams a='2', b=7.2>
     """
     params = {}
 
-    class __metaclass__(type):
+    @classmethod
+    def from_(cls, dic):
+        """
+        Build a new ParamSet from a dictionary of string-value parameters
```
Comment (ptormene) [inline openquake/commonlib/valid.py]: string-valued

File: openquake/commonlib/tests/datastore_test.py [hunk 2]
```diff
@@ -57,3 +57,10 @@ def test_hdf5(self):
 
         # it is possible to store twice the same key (work around a bug)
         self.dstore['key1'] = 'value1'
+
+    def test_parent(self):
+        self.dstore.attrs['a'] = 2
```
Comment (ptormene) [inline openquake/commonlib/tests/datastore_test.py]: Please add a comment

Comment (micheles) [inline openquake/commonlib/tests/datastore_test.py]: Added

File: openquake/commonlib/valid.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/valid.py]: Fixed

**Comment by ptormene** [comment]
LGTM

**Pull request merged**